### PR TITLE
[BugFix] fixed issue 34320: Added a check on whether the jdbc catalog data source supports partitions information tables

### LINKED ISSUE #34320: read jdbc (sr) catalog tablename   error
### Steps to reproduce the behavior (Required)
image

### Expected behavior (Required)
work normal as 3.1.3 
we can list tables in sr_old according to jdbc catalog 
### Real behavior (Required)
image
image

### StarRocks version (Required)
 - 3.1.4 on k8s shared data

### PR DESCRIPTION
**Pull request by WholeWorld-Timothy**
Why I'm doing:
When using SR as the data source for the jdbc catalog, the show tables command will cause an error.
What I'm doing:
Added a check on whether the jdbc catalog data source supports partitions information tables.
Fixes #34320

## What type of PR is this:

- [X] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [X] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [X] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.2
  - [X] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java [hunk 1]
```diff
@@ -67,13 +67,22 @@ public JDBCMetadata(Map<String, String> properties, String catalogName) {
             LOG.warn("{} not support yet", properties.get(JDBCResource.DRIVER_CLASS));
             throw new StarRocksConnectorException(properties.get(JDBCResource.DRIVER_CLASS) + " not support yet");
         }
+        schemaResolver.setSupportPartitionInformation(checkSupportPartitionInformation());
```
Comment (zombee0) [inline fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java:70]: Do you think merge checkAndSet together will be more clear?

Comment (WholeWorld-Timothy) [reply fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java:70]: This writing method is for the convenience of unit testing. I'll optimize this place.

File: fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java [hunk 2]
```diff
@@ -67,13 +67,23 @@ public JDBCMetadata(Map<String, String> properties, String catalogName) {
...
 
     public Connection getConnection() throws SQLException {
         return DriverManager.getConnection(properties.get(JDBCResource.URI),
                 properties.get(JDBCResource.USER), properties.get(JDBCResource.PASSWORD));
     }
 
+    public boolean checkSupportPartitionInformation() {
+        try (Connection connection = getConnection()) {
+            return schemaResolver.setSupportPartitionInformation(
+                    schemaResolver.checkSupportPartitionInformation(connection));
```
Comment (zombee0) [inline fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java:81]: do you think `schemaResolver.checkAndSetSupportPartitionInformation(connection) ` is more clear?

File: fe/fe-core/src/main/java/com/starrocks/connector/jdbc/MysqlSchemaResolver.java [hunk 3]
```diff
@@ -57,6 +60,31 @@ public Collection<String> listSchemas(Connection connection) {
...
+                        }
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            throw new StarRocksConnectorException(e.getMessage());
+        }
+        return false;
+    }
+
```
Comment (zombee0) [inline fe/fe-core/src/main/java/com/starrocks/connector/jdbc/MysqlSchemaResolver.java:87]: i guess this `connection.getMetaData().getTables(db, null, table, new String[] {"SYSTEM TABLE"})` maybe better, but i have not checked if it works

Comment (WholeWorld-Timothy) [reply fe/fe-core/src/main/java/com/starrocks/connector/jdbc/MysqlSchemaResolver.java:87]: This is a VIEW. According to the test results of MySQL 5.7, this table is in SYSTEM VIEW. In the system table of MySQL, the type of this table is also SYSTEM VIEW. However, the interface description in JDBC only includes SYSTEM TABLE, without SYSTEM VIEW. Therefore, I am not sure if this table is in SYSTEM VIEW in all versions of MySQL, but it must be in SYSTEM TABLE or SYSTEM VIEW.

File: fe/fe-core/src/main/java/com/starrocks/connector/jdbc/MysqlSchemaResolver.java [hunk 4]
```diff
@@ -57,6 +57,23 @@ public Collection<String> listSchemas(Connection connection) {
         }
     }
 
+    @Override
+    public boolean checkAndSetSupportPartitionInformation(Connection connection) {
+        String catalogSchema = "information_schema";
+        String partitionInfoTable = "partitions";
+        try (ResultSet tableSet = connection.getMetaData().getTables(catalogSchema, null, null, null)) {
```
Comment (zombee0) [inline fe/fe-core/src/main/java/com/starrocks/connector/jdbc/MysqlSchemaResolver.java:64]:
```suggestion
        try (ResultSet tableSet = connection.getMetaData().getTables(catalogSchema, null, partitionInfoTable, null)) {
```

**Review by zombee0 — COMMENTED** [review]
Left LGTM

**Review by zombee0 — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**